utils/tests/merkle_tree.rs

```diff
@@ -166,32 +166,32 @@ pub mod test {
 
         // Check situation when the number of items to insert is less than the number of items to delete
         tree_opt
-            .override_range(0, leaves_2.clone(), [0, 1, 2, 3])
+            .override_range(0, leaves_2.clone().into_iter(), [0, 1, 2, 3].into_iter())
             .unwrap();
 
         // check if the indexes for write and delete are the same
         tree_opt
-            .override_range(0, leaves_4.clone(), [0, 1, 2, 3])
+            .override_range(0, leaves_4.clone().into_iter(), [0, 1, 2, 3].into_iter())
             .unwrap();
         assert_eq!(tree_opt.get_empty_leaves_indices(), vec![]);
 
         // check if indexes for deletion are before indexes for overwriting
         tree_opt
-            .override_range(4, leaves_4.clone(), [0, 1, 2, 3])
+            .override_range(4, leaves_4.clone().into_iter(), [0, 1, 2, 3].into_iter())
             .unwrap();
         assert_eq!(tree_opt.get_empty_leaves_indices(), vec![0, 1, 2, 3]);
 
         // check if the indices for write and delete do not overlap completely
         tree_opt
-            .override_range(2, leaves_4.clone(), [0, 1, 2, 3])
+            .override_range(2, leaves_4.clone().into_iter(), [0, 1, 2, 3].into_iter())
             .unwrap();
         assert_eq!(tree_opt.get_empty_leaves_indices(), vec![0, 1]);
     }
 
     #[test]
     fn test_subtree_root() {
         let depth = 3;
-        let nof_leaves: usize = 6;
+        let nof_leaves: usize = 4;
```

Comment:
let's return to 6 leaves, it makes sense too show that we have up to 8 leaves on depth 3